File: go/hamming/hamming.go
package hamming

func Distance(strandA, strandB string) (mutations int) {
	for i := 0; i < len(strandA) && i < len(strandB); i++ {
		if strandA[i] != strandB[i] {
			mutations++
		}
	}
	return
}

File: javascript/hamming/hamming.js
var Hamming = {};

Hamming.compute = function(baseStrand, copyStrand){
  var distance = 0;
  for (var i = 0; i < baseStrand.length; i++) {
    if (notEqual(baseStrand[i], copyStrand[i])) {
      distance++;
    }
  }
  return distance;
};

var notEqual = function(base, copy) {
  return copy && (base != copy);
};

module.exports = Hamming;
